src/main/java/bob/Parser.java

```diff
@@ -0,0 +1,158 @@
+package bob;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+import bob.command.AddTaskCommand;
+import bob.command.ByeCommand;
+import bob.command.Command;
+import bob.command.DeleteCommand;
+import bob.command.FindCommand;
+import bob.command.ListCommand;
+import bob.command.MarkCommand;
+import bob.command.UnmarkCommand;
+import bob.exceptions.EmptyArgumentException;
+import bob.exceptions.InvalidInputException;
+import bob.exceptions.InvalidTaskNumberException;
+import bob.exceptions.MissingArgumentException;
+import bob.tasks.Deadline;
+import bob.tasks.EventTask;
+import bob.tasks.ToDo;
+
+
+/**
+ * Handles the parsing of string into commands and task objects
+ */
+public class Parser {
+
+    /**
+     * Parses the given user input into an appropriate command
+     *
+     * @param userCommand String of the user command entered
+     * @return Command object based on the input from the user
+     * @throws EmptyArgumentException If any of the required argument is empty
+     * @throws MissingArgumentException If user does not include a required argument in the command
+     * @throws InvalidInputException If user inputs an unrecognized command
+     * @throws InvalidTaskNumberException If user inputs an invalid Task Number
+     */
+    public static Command parseCommand(String userCommand)
+            throws EmptyArgumentException, MissingArgumentException, InvalidInputException, InvalidTaskNumberException {
+        Scanner scanner = new Scanner(userCommand);
+        String input = scanner.next();
+        Command command = switch (input) {
+        case "bye" -> new ByeCommand();
+        case "list" -> new ListCommand();
+        case "mark" -> new MarkCommand(scanner.nextInt() - 1);
+        case "unmark" -> new UnmarkCommand(scanner.nextInt() - 1);
+        case "todo" -> new AddTaskCommand(Parser.newToDo(scanner.nextLine().trim()));
+        case "deadline" -> new AddTaskCommand(Parser.newDeadline(scanner.nextLine().trim()));
+        case "event" -> new AddTaskCommand(Parser.newEvent(scanner.nextLine().trim()));
+        case "delete" -> new DeleteCommand(scanner.nextInt() - 1);
+        case "find" -> new FindCommand(scanner.nextLine().trim());
```

Comment:
I found this to be pretty interesting and well done, having separate classes for each command really makes the code so much more cleaner and I think really brings out encapsulation. I've noticed that you did this in many of the other classes as well. Your code follows the OOP principles really well!